packages/editor/src/components/block-list/block.js

```diff
@@ -776,7 +774,25 @@ const applyWithDispatch = withDispatch( ( dispatch, ownProps, { select } ) => {
 	};
 } );
 
+const getBlock = memize( ( name, attributes ) => {
+	return {
+		name,
+		attributes,
+	};
+} );
+
 export default compose(
+	// This withSelect is only necessary for backward compatibility
+	// Users of the editor.BlockListBlock filter used to be able to access the block prop
+	// Ideally these blocks would rely on the clientId prop only.
+	withSelect( ( select, { clientId } ) => {
+		const { getBlockName, getBlockAttributes } = select( 'core/editor' );
+		const name = getBlockName( clientId );
+		const attributes = getBlockAttributes( clientId );
+		return {
+			block: getBlock( name, attributes ),
```

Comment:
To note that there is still some risk because getBlock selector returned innerBlocks and now the block object does not contain innerBlocks. I'm not sure about what the best approach would be as calling getBlock selector would reduce the performance gain we had. But developers using the filter may find strange that this block object does not contain the innerBlocks property.

Reply:
Yes, the idea here should be to mark this prop as deprecated, I'm not sure how we can do that exactly. But if people want to full object or anything else, they should rely on `clientId` instead.